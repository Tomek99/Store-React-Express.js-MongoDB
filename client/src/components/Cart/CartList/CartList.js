import React, { useContext } from "react";
import { Context } from "../../../Contexts/Context";
import { ImBin } from "react-icons/im";
import styles from "./CartList.module.scss";
import PropTypes from "prop-types";
import BtnDeleteProduct from "../../Buttons/BtnDeleteProduct/BtnDeleteProduct";

function BasketList({ item, imBinId }) {
  const { _id, name, imageUrl, price, oldPrice, quantity } = item;

  const { deleteItem } = useContext(Context);
  return (
    <div className={styles.SingleProduct} key={_id}>
      <img src={imageUrl} alt={name} />
      <div className={styles.productDetails}>
        <h2>{name}</h2>
        <p>
          <span className={styles.newPrice}>${price} </span>
          <span className={styles.oldPrice}>
            {" "}
            {Boolean(oldPrice) ? `$${oldPrice}` : null}
          </span>
        </p>
      </div>
      <span>{quantity}</span>
      <BtnDeleteProduct
        handleBtn={deleteItem}
        id={_id}
        price={price}
        oldPrice={oldPrice}
        imBinId={imBinId}
      />
    </div>
  );
}

BasketList.propTypes = {
  id: PropTypes.number,
  name: PropTypes.string,
  imageUrl: PropTypes.string,
  newPrice: PropTypes.number,
  oldPrice: PropTypes.number,
  quantity: PropTypes.number,
  deleteItem: PropTypes.func,
};
export default BasketList;
