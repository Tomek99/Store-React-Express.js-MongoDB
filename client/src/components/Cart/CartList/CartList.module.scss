.SingleProduct {
    display: flex;
    flex-direction: row;
    justify-content: space-around;
    align-items: center;
    width: 90%;
    font-size: 1.4rem;

    img {
        width: 100%;
        max-width: 70px;
        max-height: 110px;
    }

    .productDetails {
        display: flex;
        flex-direction: column;
        justify-content: center;
        align-items: center;
        gap: 10px;
        width: 50%;
        font-weight: 300;

        h2 {
            font-size: 1.5rem;
            text-align: center;
        }

        .oldPrice {
            font-size: 1.2rem;
            color: #ccc;
            text-decoration: line-through;
        }
    }

    .btnDeleteProduct {
        align-self: flex-end;
        display: flex;
        flex-direction: column;
        align-items: center;
        justify-content: center;
        width: 35px;
        height: 35px;
        align-self: center;
        background-color: transparent;
        color: #fff;
        cursor: pointer;
    }

    .btnDeleteProduct:hover {
        border-radius: 50%;
        background-color: #434357;
    }
}
