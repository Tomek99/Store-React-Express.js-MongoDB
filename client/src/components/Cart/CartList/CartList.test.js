import { render } from "@testing-library/react";
import CartList from "./CartList";

const data = {
  available: true,
  brand: "lavazza",
  createdAt: "2023-02-02T18:36:52.221Z",
  imageUrl:
    "https://res.cloudinary.com/dvoduabha/image/upload/v1679902516/coffee/lavazza/beans/2_CaffeCremaDolce1kg_ovwzh9.png",
  intensity: 5,
  name: "Caffe Crema Dolce, 1 kg",
  oldPrice: 0,
  origin: "Mexico",
  price: "15.50",
  quantity: 2,
  rate: 5,
  type: "coffe beans",
  updatedAt: "2023-02-02T18:36:52.221Z",
  weight: "1 kg",
  __v: 0,
  _id: "63dc02c4c2a0e09b2d62f8e8",
};

test("should render cartlist component", () => {
  const { container } = render(<CartList item={data} />);

  expect(container).toMatchSnapshot();
});
