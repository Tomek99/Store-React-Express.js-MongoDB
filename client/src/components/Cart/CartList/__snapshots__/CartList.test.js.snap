// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render cartlist component 1`] = `
<div>
  <div
    class="SingleProduct"
  >
    <img
      alt="Caffe Crema Dolce, 1 kg"
      src="https://res.cloudinary.com/dvoduabha/image/upload/v1679902516/coffee/lavazza/beans/2_CaffeCremaDolce1kg_ovwzh9.png"
    />
    <div
      class="productDetails"
    >
      <h2>
        Caffe Crema Dolce, 1 kg
      </h2>
      <p>
        <span
          class="newPrice"
        >
          $
          15.50
           
        </span>
        <span
          class="oldPrice"
        >
           
        </span>
      </p>
    </div>
    <span>
      2
    </span>
    <button
      class="BtnDeleteProduct"
    >
      <svg
        fill="currentColor"
        height="18"
        stroke="currentColor"
        stroke-width="0"
        version="1.1"
        viewBox="0 0 16 16"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M2 5v10c0 0.55 0.45 1 1 1h9c0.55 0 1-0.45 1-1v-10h-11zM5 14h-1v-7h1v7zM7 14h-1v-7h1v7zM9 14h-1v-7h1v7zM11 14h-1v-7h1v7z"
        />
        <path
          d="M13.25 2h-3.25v-1.25c0-0.412-0.338-0.75-0.75-0.75h-3.5c-0.412 0-0.75 0.338-0.75 0.75v1.25h-3.25c-0.413 0-0.75 0.337-0.75 0.75v1.25h13v-1.25c0-0.413-0.338-0.75-0.75-0.75zM9 2h-3v-0.987h3v0.987z"
        />
      </svg>
    </button>
  </div>
</div>
`;
